apps/ocp-plugin/src/components/Devices/DeviceDetailsPage.tsx

```diff
@@ -1,8 +1,70 @@
 import * as React from 'react';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import { Icon, Popover, Spinner, Stack, StackItem } from '@patternfly/react-core';
+import { useAppContext } from '@flightctl/ui-components/src/hooks/useAppContext';
+import { useTranslation } from '@flightctl/ui-components/src/hooks/useTranslation';
 import DeviceDetails from '@flightctl/ui-components/src/components/Device/DeviceDetails/DeviceDetailsPage';
+import WithTooltip from '@flightctl/ui-components/src/components/common/WithTooltip';
+import { getWatchK8sResourceResult, isMicroShiftCluster } from '../../utils/clusters';
+import { ManagedCluster } from '../../types/k8s';
+
+type K8sWatchResourceError = string | object;
 
 const DeviceDetailsPage = () => {
-  return <DeviceDetails />;
+  const { t } = useTranslation();
+  const {
+    router: { useParams, Link },
+  } = useAppContext();
+  const { deviceId: clusterName } = useParams() as { deviceId: string };
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const [mc, loaded, k8sError] = useK8sWatchResource<ManagedCluster>({
+    groupVersionKind: {
+      kind: 'ManagedCluster',
+      group: 'cluster.open-cluster-management.io',
+      version: 'v1',
+    },
+    name: clusterName,
+  });
+
+  const watchResultError = getWatchK8sResourceResult(k8sError as K8sWatchResourceError, true);
+  if (k8sError && !watchResultError) {
+    return <DeviceDetails />;
```

Comment:
This is a bit weird. When we go to the details page we will see `Microshift cluster` item which is in loading state. And if it is not found, the whole item disappears. IMO we should have at at all times, and if we get 404, just show `-`

Reply:
Nir had mentioned that we shouldn't show the column when there's no MicroShift cluster.
So probably we shouldn't show the loader either, WDYT?

Reply:
Based on further discussion with Nir, we won't use the Spinner. 
And when there isn't an associated MicroShift cluster, we won't show the column either.